pandas/core/ops.py

```diff
@@ -877,11 +871,10 @@ def na_op(x, y):
                     y = _ensure_object(y)
                     result = lib.vec_binop(x, y, op)
             else:
+                # let null fall thru
+                if not isna(y):
+                    y = bool(y)
                 try:
-
-                    # let null fall thru
-                    if not isna(y):
-                        y = bool(y)
```

Comment:
This is just moving the `not isna(y)` outside of the try/except block to be more specific about whats being tried.